test/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJmap.java

```diff
@@ -0,0 +1,183 @@
+		List<String> jmapOutput = runCommand(Arrays.asList(myId, "-histo:live")); //$NON-NLS-1$
+		logJmapOutput(jmapOutput);
+		checkHeapStats(jmapOutput);
+	}
+
+	@Test
+	public void testHeapHisto() throws AttachNotSupportedException, IOException {
+		TargetManager tgtMgr = new TargetManager(TARGET_VM_CLASS, null);
+		assertTrue(AttachApiTest.CHILD_PROCESS_DID_NOT_LAUNCH, tgtMgr.syncWithTarget());
+		VirtualMachine vm = VirtualMachine.attach(tgtMgr.targetId);
+		assertTrue(vm instanceof OpenJ9VirtualMachine, "Wrong attach API VirtualMachine"); //$NON-NLS-1$
+		OpenJ9VirtualMachine myVm = (OpenJ9VirtualMachine) vm;
+		InputStream result = myVm.heapHisto();
+		checkForProperties(result);
+		result = myVm.heapHisto("-live"); //$NON-NLS-1$
+		checkForProperties(result);
+		result = myVm.heapHisto("-all"); //$NON-NLS-1$
+		checkForProperties(result);
+	}
+
+	@BeforeSuite
+	protected void setupSuite() {
+		assertTrue("This test is valid only on OpenJ9", //$NON-NLS-1$
+				System.getProperty("java.vm.vendor").contains("OpenJ9")); //$NON-NLS-1$//$NON-NLS-2$
+		getJdkUtilityPath(JMAP_COMMAND);
+		myId = TargetManager.getVmId();
+		assertNotNull(myId, "Attach ID missing"); //$NON-NLS-1$
+		log(myId);
+		
+		/* make test objects */
+		hws = new ArrayList<>();
+		for (int i = 0; i < 3; i++) {
+			hws.add(new HelloWorld());
+		}
+		objectArray = new String[][] {{"a", "b"}, {"c", "d"}}; //$NON-NLS-1$//$NON-NLS-2$ //$NON-NLS-3$//$NON-NLS-4$
+		objectVector = new Integer[] {Integer.valueOf(10), Integer.valueOf(20)};
+		primitiveArray = new int[][] {{1, 2}, {3, 4}};
+		primitiveVector = new long[] {5, 6};
+	}
+
+	private void checkHeapStats(List<String> heapStats) {
+		String hwClassName = HelloWorld.class.getName();
+		String expectedClasses[] = new String[] {hwClassName, 
+				primitiveArray.getClass().getName(),
+				primitiveVector.getClass().getName(),
+				objectArray.getClass().getName(),
+				objectVector.getClass().getName(),
+		};
+		for (String className: expectedClasses) {
+			Optional<String> result = searchSubstring(className, heapStats);
+			assertTrue(result.isPresent(), "stats missing for " + className); //$NON-NLS-1$
+		}
+		Optional<String> result = searchSubstring(hwClassName, heapStats);
+		assertTrue(result.isPresent(), hwClassName + " missing"); //$NON-NLS-1$
+		String stringResult = result.get();
+		assertTrue(stringResult.matches(".*\\b3\\b.*"), //$NON-NLS-1$
+				"wrong number of HelloWorld objects: " + stringResult); //$NON-NLS-1$
+	}
+
+	private void checkForProperties(InputStream result) throws IOException {
+		try (BufferedReader jpsOutReader = new BufferedReader(new InputStreamReader(result))) {
+			List<String> outputLines = jpsOutReader.lines().collect(Collectors.toList());
+			String propertiesName = Properties.class.getName();
+			if (!searchSubstring(propertiesName, outputLines).isPresent()) {
+				log(propertiesName + " not found in the following data: "); //$NON-NLS-1$
+				for (String l: outputLines) {
```

Comment:
Formatting: space around `:`.
The name `l` is a particularly bad choice for a variable name; in some fonts it looks too much like `1` and it is not very descriptive: how about `outputLine`?

Reply:
Good idea.  I used `opLine` for brevity.

Reply:
Brevity was part of the problem; that's better, but since the compiler doesn't care why make it harder on our future selves reading this code?